stubs/common/EloquentBuilder.stub

```diff
@@ -544,7 +544,19 @@ class Builder
     public function chunk($count, $callback);
 }
 
-class Scope {}
+/**
+ * @template TModelClass of Model
+ */
+class Scope {
+    /**
+     * Apply the scope to a given Eloquent query builder.
+     *
+     * @param  \Illuminate\Database\Eloquent\Builder<TModelClass>  $builder
+     * @param  TModelClass  $model
+     * @return void
+     */
+    public function apply(Builder $builder, Model $model);
+}
```

Comment:
This looks brittle to me:

```suggestion
/**
 * @template TBuilder of Builder
 * @template TModelClass of Model
 */
class Scope {
    /**
     * Apply the scope to a given Eloquent query builder.
     *
     * @param  TBuilder  $builder
     * @param  TModelClass  $model
     * @return void
     */
    public function apply(Builder $builder, Model $model);
}
```


```php
/** @extends Scope<Builder<Update|Announcement>, Update|Annoucement> */
class AnnouncementTypeScope implements Scope
{
    public function apply(Builder $builder, Model $model): void
    {
        ....
    }
}
```

I'd also recommend moving to it's own stub now that you're adding generics to it